File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {

  public static void main(String[] args) throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    String[] nums = br.readLine().split(" ");

    int a = Integer.parseInt(nums[0]);
    int b = Integer.parseInt(nums[1]);

    int gcd = gcd(a, b);

    System.out.println(gcd);
    System.out.println(a * b / gcd);
  }

  public static int gcd(int a, int b) {
    if (b == 0) {
      return a;
    } else {
      return gcd(b, a % b);
    }
  }
}

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {
    val token = StringTokenizer(readLine())

    val a = token.nextToken().toInt()
    val b = token.nextToken().toInt()

    val gcd = gcd(a, b);
    println(gcd)
    println(a * b / gcd)


}

fun gcd(a: Int, b: Int): Int = if (b == 0) a else gcd(b, a % b)
